packages/crayons-core/src/components/data-table/data-table.tsx

```diff
@@ -390,6 +369,318 @@ export class DataTable {
+      currentElementIndex,
+      otherElements.length
+    );
+    if (modifiedCurrentElement.position !== currentElement.position) {
+      preElements.map((preElement) => {
+        if (
+          preElement.position >= modifiedCurrentElement.position &&
+          preElement.position <= currentElement.position
+        ) {
+          preElement.position = preElement.position + 1;
+        }
+        return preElement;
+      });
+      postElements.map((postElement) => {
+        if (
+          postElement.position <= modifiedCurrentElement.position &&
+          postElement.position >= currentElement.position
+        ) {
+          postElement.position = postElement.position - 1;
+        }
+        return postElement;
+      });
+    }
+    const newStateOrder = [
+      ...preElements,
+      modifiedCurrentElement,
+      ...postElements,
+    ].sort((a, b) => a.position - b.position);
+    return newStateOrder;
+  }
+
+  /**
+   * Private
+   * lockFocusInside lock the focus inside modal overlay
+   */
+  lockFocusInsideSettings() {
+    const resetFocus = (event) => {
+      event.stopPropagation();
+      if (event.shiftKey === false && event.key === 'Tab') {
+        this.settingsResetButton.focus();
+      }
+    };
+    const updateFocus = (event) => {
+      event.stopPropagation();
+      if (event.shiftKey === true && event.key === 'Tab') {
+        this.settingsUpdateButton.setFocus();
+      }
+    };
+    this.settingsUpdateButton.addEventListener('keydown', resetFocus);
+    this.settingsResetButton.addEventListener('keydown', updateFocus);
+  }
+
+  /**
+   * Private
+   * toggleSettings
+   * @param isOpen opens/closes the table
+   * @return isOpen current open/close state
+   */
+  async toggleSettings(isOpen = true) {
+    await this.resetSettings();
+    this.isSettingsOpen = isOpen;
+    await this.waitForNextRender();
+    this.popperInstance.update();
+    if (isOpen) {
+      this.settingsInput.setFocus();
+      this.lockFocusInsideSettings();
+    }
+    return isOpen;
+  }
+
+  /**
+   * resetSettings function to reset the temporary column state in settings using table column state.
+   */
+  async resetSettings() {
+    this.settingSearchText = '';
+    this.columnsDragSetting = [];
+    this.columnsHideSetting = [];
+    this.orderedColumns.forEach((column) => {
+      const columnInfo: any = {};
+      columnInfo.key = column.key;
+      columnInfo.text = column.text;
+      columnInfo.position = column.position;
+      columnInfo.hide = column.hide || false;
+      this.columnsDragSetting = [...this.columnsDragSetting, columnInfo];
+    });
+    this.columns.forEach((column) => {
+      const columnInfo: any = {};
+      columnInfo.key = column.key;
+      columnInfo.text = column.text;
+      columnInfo.hide = this.orderedColumns.filter(
+        (orderedColumn) => orderedColumn.key === column.key
+      )[0].hide;
+      this.columnsHideSetting = [...this.columnsHideSetting, columnInfo];
```

Comment:
instead of updating the state on each iteration of the `forEach`, can we use map and return an array -`[columnInfo]` and then merge it with the state like 
`this.columnsHideSetting = [...this.columnsHideSetting, ...columnsInfo]`

Reply:
Changed as suggested.